velox/functions/prestosql/tests/CustomTypeCastTest.cpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cstdint>
+#include <limits>
+
+#include "velox/core/Expressions.h"
+#include "velox/core/ITypedExpr.h"
+#include "velox/expression/VectorFunction.h"
+#include "velox/functions/prestosql/tests/FunctionBaseTest.h"
+#include "velox/functions/prestosql/types/JsonType.h"
+
+using namespace facebook::velox;
+
+namespace {
+
+/// Returns indices buffer with sequential values going from size - 1 to 0.
+BufferPtr makeIndicesInReverse(vector_size_t size, memory::MemoryPool* pool) {
+  auto indices = allocateIndices(size, pool);
+  auto rawIndices = indices->asMutable<vector_size_t>();
+  for (auto i = 0; i < size; i++) {
+    rawIndices[i] = size - 1 - i;
+  }
+  return indices;
+}
+
+/// Wraps input in a dictionary that reverses the order of rows.
+class TestingDictionaryFunction : public exec::VectorFunction {
+ public:
+  bool isDefaultNullBehavior() const override {
+    return false;
+  }
+
+  void apply(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      const TypePtr& /* outputType */,
+      exec::EvalCtx* context,
+      VectorPtr* result) const override {
+    VELOX_CHECK(rows.isAllSelected());
+    const auto size = rows.size();
+    auto indices = makeIndicesInReverse(size, context->pool());
+    *result = BaseVector::wrapInDictionary(
+        BufferPtr(nullptr), indices, size, args[0]);
+  }
+
+  static std::vector<std::shared_ptr<exec::FunctionSignature>> signatures() {
+    // T -> T
+    return {exec::FunctionSignatureBuilder()
+                .typeVariable("T")
+                .returnType("T")
+                .argumentType("T")
+                .build()};
+  }
+};
+
+} // namespace
+
+class CastExprTest : public functions::test::FunctionBaseTest {
+ protected:
+  CastExprTest() {
+    exec::registerVectorFunction(
+        "testing_dictionary",
+        TestingDictionaryFunction::signatures(),
+        std::make_unique<TestingDictionaryFunction>());
+  }
+
+  template <typename TTo>
+  void evaluateCustomCast(
+      const TypePtr& fromType,
+      const TypePtr& toType,
+      const RowVectorPtr& input,
+      const VectorPtr& expected,
+      bool tryCast = false) {
```

Comment:
Looks like there are no callers that pass tryCast =  true. Remove?

Reply:
Removed. Thank you!